h2o-genmodel/src/main/java/hex/genmodel/algos/tree/Graph.java

```diff
@@ -0,0 +1,54 @@
+package hex.genmodel.algos.tree;
+
+import java.io.PrintStream;
+import java.util.ArrayList;
+
+/**
+ * Graph for representing a forest.
+ */
+public class Graph {
+  private ArrayList<Subgraph> subgraphArray = new ArrayList<>();
+  private int nextSubgraphNumber = 0;
+
+  public Graph() {
+    subgraphArray = new ArrayList<>();
+  }
+
+  Subgraph makeSubgraph(String name) {
+    Subgraph sg = new Subgraph(nextSubgraphNumber, name);
```

Comment:
Suggestion: replace `nextSubgraphNumber` with `subgraphArray.size()`. Then the `nextSubgraphNumber` can be removed completely, simplifying the class